src/lib/continued_fraction.ml

```diff
@@ -0,0 +1,36 @@
+let small = 1e-50
+let max_iteration_default = 200
+
+type t = { get_a : int -> float -> float; get_b : int -> float -> float }
+
+let init ~get_a ~get_b = {get_a;get_b}
+
+let evaluate t ?(epsilon=small) ?(max_iterations=max_iteration_default) x =
+  let a = t.get_a 0 0.0 in
+  let hPrev = 
+    (* use the value of small as epsilon criteria for zero checks *)
+    if (Util.significantly_different_from ~d:small a 0.0) then a else small in
+  let rec loop dPrev cPrev hPrev n : float =
+    if n > max_iterations then raise 
+      (Invalid_argument "reached max iterations") else
```

Comment:
Agreed, but we probably need a commonly shared exceptions module to keep things consistent. Also, I can probably eventually factor out your existing ad-hoc version of the continued fraction that exists already in the ```Functions``` module.